test/extensions/http/header_validators/envoy_default/header_validator_test.cc

```diff
@@ -16,80 +9,281 @@ namespace HeaderValidators {
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateSchemeHeader(invalid_first_char),
+                             UhvResponseCodeDetail::get().InvalidScheme);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusNoneValid) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::WholeNumber;
+  HeaderString valid{"200"};
+  HeaderString valid_outside_of_range{"1024"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_TRUE(uhv->validateStatusHeader(mode, valid).ok());
+  EXPECT_TRUE(uhv->validateStatusHeader(mode, valid_outside_of_range).ok());
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusNoneInvalid) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::WholeNumber;
+  HeaderString invalid{"asdf"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusRangeValid) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::ValueRange;
+  HeaderString valid{"200"};
+  HeaderString invalid_max{"1024"};
+  HeaderString invalid_min{"99"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_TRUE(uhv->validateStatusHeader(mode, valid).ok());
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_max),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_min),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusRangeInvalidMin) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::ValueRange;
+  HeaderString invalid_min{"99"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_min),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusRangeInvalidMax) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::ValueRange;
+  HeaderString invalid_max{"1024"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_max),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateResponseStatusOfficalCodes) {
+  auto mode = BaseHttpHeaderValidator::StatusPseudoHeaderValidationMode::OfficialStatusCodes;
+  HeaderString valid{"200"};
+  HeaderString invalid_unregistered{"420"};
+  auto uhv = createBase(empty_config);
+
+  EXPECT_TRUE(uhv->validateStatusHeader(mode, valid).ok());
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateStatusHeader(mode, invalid_unregistered),
+                             UhvResponseCodeDetail::get().InvalidStatus);
+}
+
+TEST_F(BaseHeaderValidatorTest, ValidateGenericHeaderNameValid) {
```

Comment:
For header name and value validity tests I would use an approach similar to https://github.com/envoyproxy/envoy/blob/5163f3e16cd774a63e833c9f3b3a07a6da9f0fc0/test/common/http/http2/codec_impl_test.cc#L4283 and actually change one the characters in name/value through the entire range.